.github/workflows/integration-test.yml

```diff
@@ -80,12 +96,28 @@ jobs:
           java-version: 17
       - name: Tools Versions
         run: ./scripts/ci_steps/log_tool_versions.sh
+      - name: Add Credentials to Test Plan
+        run: |
+          # JSON-escape the credentials.  They are also surrounded with double quotes.
+          AKID_ESCAPED=`echo -n "$AWS_ACCESS_KEY_ID" | jq -Rsa .`
+          SECRET_ESCAPED=`echo -n "$AWS_SECRET_ACCESS_KEY" | jq -Rsa .`
+          REGION_ESCAPED=`echo -n "$AWS_DEFAULT_REGION" | jq -Rsa .`
+          TOKEN_ESCAPED=`echo -n "$AWS_SESSION_TOKEN" | jq -Rsa .`
+          # Insert the credentials into the .xctestplan file, write the modified JSON
+          # to a temp file, then move the temp over the original.
+          jq ".defaultOptions.environmentVariableEntries += [{\"key\": \"AWS_ACCESS_KEY_ID\", \"value\": $AKID_ESCAPED}, {\"key\": \"AWS_SECRET_ACCESS_KEY\", \"value\": $SECRET_ESCAPED}, {\"key\": \"AWS_DEFAULT_REGION\", \"value\": $REGION_ESCAPED}, {\"key\": \"AWS_SESSION_TOKEN\", \"value\": $TOKEN_ESCAPED}]" XCTestPlans/AWSIntegrationTestsOnCI.xctestplan > testplan.tmp
+          mv testplan.tmp XCTestPlans/AWSIntegrationTestsOnCI.xctestplan
```

Comment:
Above is the "secret sauce": to get AWS credentials into the Xcode simulators: use the `jq` command-line utility to insert the credentials into the scheme's `.xctestplan` file.

(`.xctestplan` is a JSON file used by Xcode to conbfigure tests, and `jq` is a CLI utility for reading, writing & updating JSON.)

Reply:
I don't believe jq comes pre-installed on Macs unless github actions makes it available, does this work when run on a github actions environment?

For a script I wrote to run on Amazon Linux I built-in a bash script method that checks if jq is installed and installs it if not, you may need to do something similar here

Reply:
NVM I see it comes with github actions